lib/babe/epoch.go

```diff
@@ -184,11 +188,14 @@ func (b *Service) getLatestEpochData() (resEpochData *epochData, error error) {
 func (b *Service) getFirstAuthoringSlot(epoch uint64, epochData *epochData) (uint64, error) {
 	startSlot := getCurrentSlot(b.constants.slotDuration)
 	for i := startSlot; i < startSlot+b.constants.epochLength; i++ {
-		_, err := b.runLottery(i, epoch, epochData)
+		_, err := claimSlot(epoch, i, epochData, b.keypair)
 		if err != nil {
 			if errors.Is(err, errOverPrimarySlotThreshold) {
 				continue
 			}
+			if errors.Is(err, errNotOurTurnToPropose) {
+				continue
+			}
 			return 0, fmt.Errorf("error running slot lottery at slot %d: error %w", i, err)
```

Comment:
this nested if can be removed.

```suggestion
	if errors.Is(err, errOverPrimarySlotThreshold) {
		continue
	}
	if errors.Is(err, errNotOurTurnToPropose) {
		continue
	} else if err != nil {
			      return 0, fmt.Errorf("error running slot lottery at slot %d: error %w", i, err)
	}			
```